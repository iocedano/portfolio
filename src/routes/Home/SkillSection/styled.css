.skillsection-container {
  width: 80vw;
  margin-top: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: auto;
}

.skillsection-header {
  font-size: 3rem;
  line-height: 20px;
}

.skillsection-list {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  margin-top: 50px;
  grid-gap: 20px;
  list-style-type: none;
}
