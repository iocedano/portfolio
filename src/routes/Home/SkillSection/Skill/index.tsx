import React, { useEffect, useState, memo } from 'react';
import PropTypes, { InferProps } from 'prop-types';
import validIcons from '../../../../config/validicons.json';
import './styled.css';

const SkillPropsTypes = {
  type: PropTypes.oneOf(validIcons.list).isRequired,
  description: PropTypes.string
};

export type SkillType = InferProps<typeof SkillPropsTypes>;

const Skill = ({ type, description } : SkillType) => {
  const [img, setImg] = useState('');

  useEffect(() => {
    import(`../../../../assets/icons/${type}.svg`).then((image) => {
      setImg(image.default);
    });
  }, [type]);

  return (
    <li className="skill-item">
      <div className="skill-item-img-content">
        <img className="skill-item-img" src={img} alt={`skill know job with require ${type} developer`} />
      </div>
      {description && (
        <p className="skill-item-description">{description}</p>
      )}
    </li>
  );
};

Skill.propTypes = SkillPropsTypes;

Skill.defaultProps = {
  description: undefined
}

export default memo(Skill);
