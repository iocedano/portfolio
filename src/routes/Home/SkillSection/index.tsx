import React from 'react';
import PropTypes, { InferProps } from 'prop-types';
import Skill, { SkillType } from './Skill';
import './styled.css';

const SkillSectionPropTypes = {
  skills: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.string,
    description: PropTypes.string
  }))
};

type SkillSectionType = InferProps<typeof SkillSectionPropTypes | { skills: [SkillType] }>;

const SkillSection = ({ skills } : SkillSectionType) => (
  <section className="skillsection-container">
    <h2 className="skillsection-header">Skills</h2>
    <ul className="skillsection-list">
      {Array.isArray(skills) && skills.map(({type, description}) => (
        <Skill key={type} type={type} description={description} />
      ))}
    </ul>
  </section>
);

SkillSection.propTypes = SkillSectionPropTypes;

SkillSection.defaultProps = {
  skills: []
};

export default SkillSection;
