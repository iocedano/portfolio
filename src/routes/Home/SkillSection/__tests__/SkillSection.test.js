import React from 'react';
import { render, act } from '@testing-library/react';
import SkillSection from '..';

const Skills = [{ type: 'yarn' }];

describe('Home/SkillSection', () => {
  it('should render', async () => {
    const promise = Promise.resolve();
    const { container } = render(<SkillSection skills={Skills} />);

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise)
  });

  it('should not render the list if the skills is empty', () => {
    const { container } = render(<SkillSection skills={[]} />);

    expect(container.querySelector('ul li')).toBeNull();
  });

  it('should render the list if the skills', async() => {
    const promise = Promise.resolve();
    const { container } = render(<SkillSection skills={Skills} />);

    expect(container.querySelectorAll('ul li')).toHaveLength(1);
    await act(() => promise)
  });
});
