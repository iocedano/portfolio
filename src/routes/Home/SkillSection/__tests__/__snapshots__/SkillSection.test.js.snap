// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/SkillSection should render 1`] = `
<section
  class="skillsection-container"
>
  <h2
    class="skillsection-header"
  >
    Skills
  </h2>
  <ul
    class="skillsection-list"
  >
    <li
      class="skill-item"
    >
      <div
        class="skill-item-img-content"
      >
        <img
          alt="skill know job with require yarn developer"
          class="skill-item-img"
          src=""
        />
      </div>
    </li>
  </ul>
</section>
`;
